proposals/Proposal: add cookies.removeAll() method.md

```diff
@@ -0,0 +1,147 @@
+**Summary**
+
+Adds a new API `cookies.removeAll()` to allow for the removal of multiple cookies with a single call.
+
+**Document Metadata**
+
+**Author:**  [aselya](https://github.com/aselya)
+
+**Sponsoring Browser:**  Chrome
+
+**Contributors:**  
+
+**Created:**  2024-09-18
+
+**Related Issues:**  [PrivacyCG/CHIPS issue 6 - How do Partitioned cookies interact with browser extensions?](https://github.com/privacycg/CHIPS/issues/6)
+
+## Motivation
+
+Prior to the introduction of [partitioned cookies](https://github.com/privacycg/CHIPS), the required parameters (`url` and `name`) used by the existing `cookies.remove()` api would be unique to a single cookie. The inclusion of an optional `partitionKey` means that there can be more than one cookie that matches the currently required parameters for `cookies.remove()`. Because of this, handling partitioned cookies with `cookies.remove()` requires additional steps which can be simplified by the inclusion of this new API.
+
+### Objective
+
+This new API addresses two workflows for developers:
```

Comment:
It seems that the examples you provided is how extensions would have to achieve the functionality these days. Can you rephrase to emphasize that? To someone not deeply aware of the details, it can be read as part of the proposal for something net, not of existing behavior.

Reply:
Updated the language and I would appreciate feedback on if my rephrasing addresses the issue you brought up.